backend/src/main/java/com/eshka/service/impl/QueueServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.eshka.service.impl;
+
+import com.eshka.entity.Queue;
+import com.eshka.exception.QueueNotFoundException;
+import com.eshka.repository.QueueRepository;
+import com.eshka.service.QueueService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class QueueServiceImpl implements QueueService {
+    private final QueueRepository queueRepository;
+
+    @Override
+    public Queue findById(long id) {
+        return queueRepository.findById(id).orElseThrow(
+                () -> new QueueNotFoundException("queue not found"));
+    }
+
+    @Override
+    public Queue createNewQueue(Queue request) {
+        return queueRepository.save(request);
+    }
+
+    @Override
+    public Queue editQueue(Queue request) {
+        Queue queue = queueRepository.findById(request.getId()).orElseThrow(
+                () -> new QueueNotFoundException("queue not found"));
+        queue.setDescription(request.getDescription());
+        queue.setTitle(request.getTitle());
+        return queueRepository.save(queue);
+    }
```

Comment:
The logic for editing qeueu is more complex than you think.
Exept title or description, I can change the start and the end date.

Reply:
> The logic for editing qeueu is more complex than you think.
> Exept title or description, I can change the start and the end date.

but you only about title and description)